hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestObserverWithRouter.java

```diff
@@ -51,13 +51,16 @@
 import org.apache.hadoop.hdfs.server.federation.resolver.FederationNamenodeServiceState;
 import org.apache.hadoop.hdfs.server.federation.resolver.MembershipNamenodeResolver;
 import org.apache.hadoop.hdfs.server.namenode.NameNode;
+import org.apache.hadoop.hdfs.server.namenode.ha.RouterObserverReadProxyProvider;
```

Comment:
This test is not specifically checking anything on msyncs. Can we add something?

Reply:
Yes, I've just added some tests for when auto-sync is enabled. One for the always msync case, and another for when it is periodic.

The existing `testRouterMsync` tests when auto-msync is disabled.